lib/datadog/appsec/extensions.rb

```diff
@@ -0,0 +1,55 @@
+# typed: true
+require 'forwardable'
+require 'datadog/appsec/configuration'
+
+module Datadog
+  module AppSec
+    # Extends Datadog tracing with AppSec features
+    module Extensions
+      # Inject AppSec into global objects.
+      def self.activate!
+        Core::Configuration::Settings.include(Settings)
+      end
+
+      # Global Datadog configuration mixin
+      module Settings
+        # Exposes AppSec settings through the
+        # `Datadog.configure {|c| c.appsec._option_ }`
+        # configuration path.
+        def appsec
+          @appsec ||= AppSecAdapter.new(AppSec.settings)
+        end
+      end
+
+      # Merges {Datadog::AppSec::Configuration::Settings} and {Datadog::AppSec::Configuration::DSL}
+      # into a single read/write object.
+      class AppSecAdapter
+        extend Forwardable
+
+        def initialize(settings)
+          @settings = settings
+        end
+
+        # Writer methods
+        AppSec::Configuration::DSL.instance_methods(false).each do |met|
+          define_method(met) do |*arg|
```

Comment:
If it's practical to avoid defining methods dynamically like this, we should do so. (Performance cost associated with scope here I think.)

Reply:
I would rather not have dynamic methods as well if at all possible, generally favouring explicit over implicit.

Reply:
Done!